src/core/server/http/http_server.ts

```diff
@@ -147,16 +147,22 @@ export class HttpServer {
       for (const route of router.getRoutes()) {
         this.log.debug(`registering route handler for [${route.path}]`);
         // Hapi does not allow payload validation to be specified for 'head' or 'get' requests
-        const validate = ['head', 'get'].includes(route.method) ? undefined : { payload: true };
+        const validate = isSafeMethod(route.method) ? undefined : { payload: true };
```

Comment:
hapi doesn't support handlers for the `head` method https://github.com/hapijs/hapi/issues/795